dll/win32/shlwapi/utils.cpp

```diff
@@ -115,3 +133,99 @@ PathFileExistsDefExtAndAttributesW(
 
     return TRUE;
 }
+
+static inline BOOL
+SHLWAPI_IsBogusHRESULT(HRESULT hr)
+{
+    return (hr == E_FAIL || hr == E_INVALIDARG || hr == E_NOTIMPL);
+}
+
+/*************************************************************************
+ * IShellFolder_GetDisplayNameOf [SHLWAPI.316]
+ */
+EXTERN_C HRESULT WINAPI
+IShellFolder_GetDisplayNameOf(
+    _In_ IShellFolder *psf,
+    _In_ LPCITEMIDLIST pidl,
+    _In_ DWORD uFlags,
+    _Out_ LPSTRRET lpName,
+    _In_ DWORD dwRetryFlags)
+{
+    HRESULT hr;
+
+    TRACE("(%p, %p, 0x%lX, %p, 0x%lX)\n", psf, pidl, uFlags, lpName, dwRetryFlags);
+
+    hr = psf->GetDisplayNameOf(pidl, uFlags, lpName);
+    if (!SHLWAPI_IsBogusHRESULT(hr))
+        return hr;
+
+    dwRetryFlags |= 0x80000000;
+
+    if ((uFlags & SHGDN_FORPARSING) == 0)
+        dwRetryFlags |= SFGDNO_RETRYWITHFORPARSING;
+
+    /* It seems the function is actually retrying here */
+    FIXME("dwRetryFlags: 0x%X\n", dwRetryFlags);
+
+    return hr;
+}
+
+/*************************************************************************
+ * IShellFolder_ParseDisplayName [SHLWAPI.317]
+ */
+EXTERN_C HRESULT WINAPI
+IShellFolder_ParseDisplayName(
+    _In_ IShellFolder *psf,
+    _In_opt_ HWND hwndOwner,
+    _In_opt_ LPBC pbcReserved,
+    _In_ LPOLESTR lpszDisplayName,
+    _Out_opt_ ULONG *pchEaten,
+    _Out_opt_ PIDLIST_RELATIVE *ppidl,
+    _Out_opt_ ULONG *pdwAttributes)
+{
+    ULONG dummy1, dummy2;
+
+    TRACE("(%p, %p, %s, %p, %p, %p)\n", hwndOwner, pbcReserved, lpszDisplayName,
+                                        pchEaten, ppidl, pdwAttributes);
+
+    /* Improve safety */
```

Comment:
This doesn't explain anything so either remove the comment, or write something else.
This is more likely to be, that the actual IShellFolder::ParseDisplayName (or some implementers of the method)  doesn't accept optional out parameters for pchEaten and pdwAttributes, but that exported function you're implementing explicitly does.

Reply:
Deleted.